daml-assistant/daml-helper/DamlHelper.hs

```diff
@@ -121,3 +202,39 @@ installExtension src target =
          -- but medium term it probably makes more sense to add the extension to the marketplace
          -- and make it backwards compatible
          const $ pure ())
+
+-- | `waitForConnectionOnPort sleep port` keeps trying to establish a TCP connection on the given port.
+-- Between each connection request it calls `sleep`.
+waitForConnectionOnPort :: IO () -> Int -> IO ()
+waitForConnectionOnPort sleep port = do
+    let hints = defaultHints { addrFlags = [AI_NUMERICHOST, AI_NUMERICSERV], addrSocketType = Stream }
+    addr : _ <- getAddrInfo (Just hints) (Just "127.0.0.1") (Just $ show port)
+    untilJust $ do
+        r <- tryIO $ checkConnection addr
+        either (const $ Nothing <$ sleep) (const $ pure $ Just ()) r
```

Comment:
This is almost unintelligible. Can we please use a `case` expression. :D

Reply:
Changed